lib/tasks/reports.rake

```diff
@@ -61,6 +61,17 @@ namespace :reports do
   end
 
 
+  task :monthly_report,[:year, :month, :date] => [:environment] do |t, args|
+    args.with_defaults(:year => Date.today.year, :month => Date.today.month - 1, :date => 1)
+    test_date = Date.new(args.year.to_i, args.month.to_i, args.date.to_i)
```

Comment:
You could probably use beginning_of_month helper here http://api.rubyonrails.org/classes/DateAndTime/Calculations.html#method-i-beginning_of_month